xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/XWikiHibernateStore.java

```diff
@@ -2066,59 +2071,60 @@ public String getName()
             {
-                if ("logout".equals(((ActionExecutingEvent) event).getActionName())) {
-                    final XWikiContext ctx = (XWikiContext) data;
-                    if (ctx.getUserReference() != null) {
-                        releaseAllLocksForCurrentUser(ctx);
-                    }
+                UserUnauthenticatedEvent userUnauthenticatedEvent = (UserUnauthenticatedEvent) event;
+                if (userUnauthenticatedEvent.getUserReference() != null) {
+                    DocumentReference userDoc = XWikiHibernateStore.this.userReferenceSerializer.serialize(
+                        userUnauthenticatedEvent.getUserReference());
+                    releaseAllLocksForUser(userDoc, (XWikiContext) data);
                 }
             }
         });
     }
 
     /**
      * Release all of the locks held by the currently logged in user.
-     *
-     * @param ctx the XWikiContext, used to start the connection and get the user name.
      */
-    private void releaseAllLocksForCurrentUser(final XWikiContext ctx)
+    private void releaseAllLocksForUser(final DocumentReference userDoc, final XWikiContext context)
     {
         try {
-            executeWrite(ctx, session -> {
+            executeWrite(context, session -> {
                 final Query query = session.createQuery("delete from XWikiLock as lock where lock.userName=:userName");
                 // Using deprecated getUser() because this is how locks are created.
                 // It would be a maintainibility disaster to use different code paths
                 // for calculating names when creating and removing.
-                query.setParameter("userName", ctx.getUser());
+                query.setParameter("userName", this.compactWikiEntityReferenceSerializer.serialize(userDoc));
                 query.executeUpdate();
 
                 return null;
             });
         } catch (Exception e) {
             String msg = "Error while deleting active locks held by user.";
             try {
-                this.endTransaction(ctx, false);
+                this.endTransaction(context, false);
             } catch (Exception utoh) {
                 msg += " Failed to commit OR rollback [" + utoh.getMessage() + "]";
             }
             throw new UnexpectedException(msg, e);
         }
 
-        // If we're in a non-main wiki & the user is global,
-        // switch to the global wiki and delete locks held there.
-        if (!ctx.isMainWiki() && ctx.isMainWiki(ctx.getUserReference().getWikiReference().getName())) {
-            final String cdb = ctx.getWikiId();
+        if (this.wikiDescriptorManager.isMainWiki(userDoc.getWikiReference().getName())) {
             try {
-                ctx.setWikiId(ctx.getMainXWiki());
-                this.releaseAllLocksForCurrentUser(ctx);
-            } finally {
-                ctx.setWikiId(cdb);
+                String currentWiki = context.getWikiId();
```

Comment:
Would probably make more sense to use getWikiReference since the only point of what you are getting is to restore it with setWikiReference.